`apoc.custom.*` procedures don't work with database aliases

## Expected Behavior (Mandatory)

It should works correctly

## Actual Behavior (Mandatory)

It throws an error:
```
Failed to invoke procedure `apoc.custom.installFunction`: Caused by: java.lang.RuntimeException: The user database with name 'test' does not exist
```

## How to Reproduce the Problem


  1. `CREATE ALIAS test FOR DATABASE neo4j`
  1. `USE system CALL apoc.custom.installProcedure('answer2() :: (answer::INT)','RETURN 42 as answer', 'test')`


### Versions

  - OS: OSX
  - Neo4j: 2025.09.0
  - Neo4j-Apoc: 2025.09.0

--- a/extended-it/src/test/java/apoc/dv/DataVirtualizationCatalogClusterRoutingTest.java
+++ b/extended-it/src/test/java/apoc/dv/DataVirtualizationCatalogClusterRoutingTest.java
@@ -34,6 +34,7 @@ import static apoc.util.TestContainerUtil.testCallEmpty;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
+import static org.neo4j.configuration.GraphDatabaseSettings.DEFAULT_DATABASE_NAME;
 import static org.neo4j.configuration.GraphDatabaseSettings.SYSTEM_DATABASE_NAME;
 
 
@@ -65,8 +66,8 @@ public class DataVirtualizationCatalogClusterRoutingTest {
     public void testVirtualizeCSV() {
         dvInSysLeaderMemberCommon(PROCEDURE_NOT_ROUTED_ERROR, SYSTEM_DATABASE_NAME,
                 (session) -> testCall(session, APOC_DV_INSTALL_QUERY,
-                        APOC_DV_INSTALL_PARAMS,
-                        (row) -> assertCatalogContent(row, CSV_TEST_FILE)), APOC_DV_INSTALL_PARAMS
+                        getApocDvInstallParams(DEFAULT_DATABASE_NAME),
+                        (row) -> assertCatalogContent(row, CSV_TEST_FILE)), getApocDvInstallParams(DEFAULT_DATABASE_NAME)
         );
 
         clusterSession.executeRead(tx -> {
@@ -113,7 +114,7 @@ public class DataVirtualizationCatalogClusterRoutingTest {
 
         dvInSysLeaderMemberCommon(PROCEDURE_NOT_ROUTED_ERROR, SYSTEM_DATABASE_NAME,
                 (session) -> testCallEmpty(session, APOC_DV_DROP_QUERY,
-                            APOC_DV_DROP_PARAMS), APOC_DV_DROP_PARAMS
+                            getApocDvDropParams(DEFAULT_DATABASE_NAME)), getApocDvDropParams(DEFAULT_DATABASE_NAME)
         );
 
     }

--- a/extended-it/src/test/java/apoc/dv/DataVirtualizationCatalogMultiDbTest.java
+++ b/extended-it/src/test/java/apoc/dv/DataVirtualizationCatalogMultiDbTest.java
@@ -1,0 +1,154 @@
+package apoc.dv;
+
+import apoc.util.Neo4jContainerExtension;
+import apoc.util.TestContainerUtil;
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.neo4j.driver.Record;
+import org.neo4j.driver.*;
+import org.neo4j.driver.types.Node;
+import org.neo4j.driver.types.Path;
+import org.neo4j.driver.types.Relationship;
+
+import java.io.File;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static apoc.dv.DataVirtualizationCatalogTestUtil.*;
+import static apoc.util.ExtendedTestContainerUtil.singleResultFirstColumn;
+import static apoc.util.MapUtil.map;
+import static apoc.util.TestContainerUtil.*;
+import static org.junit.Assert.assertEquals;
+import static org.neo4j.configuration.GraphDatabaseSettings.DEFAULT_DATABASE_NAME;
+import static org.neo4j.configuration.GraphDatabaseSettings.SYSTEM_DATABASE_NAME;
+import static org.neo4j.test.assertion.Assert.assertEventually;
+
+public class DataVirtualizationCatalogMultiDbTest {
+
+    private static Neo4jContainerExtension neo4jContainer;
+    private static Driver driver;
+    private static Session neo4jSession;
+    private static Session fooSession;
+    private static Session systemSession;
+
+    private static final String DB_FOO = "dbfoo";
+
+    @BeforeClass
+    public static void setupContainer() throws Exception {
+        neo4jContainer = createEnterpriseDB(
+                List.of(TestContainerUtil.ApocPackage.EXTENDED, TestContainerUtil.ApocPackage.CORE), true
+        );
+        neo4jContainer.start();
+        driver = neo4jContainer.getDriver();
+
+        FileUtils.copyFileToDirectory(new File(new URI(FILE_URL).toURL().getPath()), importFolder);
+        
+        createDatabases();
+        createSessions();
+    }
+
+    @After
+    public void cleanDb() {
+        neo4jSession.executeWrite(tx -> tx.run("MATCH (n) DETACH DELETE n;").consume());
+        fooSession.executeWrite(tx -> tx.run("MATCH (n) DETACH DELETE n;").consume());
+    }
+
+    @Test
+    public void testVirtualizeCSVWithDefaultDb() {
+        testVirtualizeCSVCommon(DEFAULT_DATABASE_NAME, neo4jSession);
+    }
+
+    @Test
+    public void testVirtualizeCSVWithSecondaryDb() {
+        testVirtualizeCSVCommon(DB_FOO, fooSession);
+    }
+
+    @Test
+    public void testVirtualizeCSVWithSecondaryDbUsingAnAlias() {
+        testVirtualizeCSVCommon("test-alias", fooSession);
+    }
+
+    private static void testVirtualizeCSVCommon(String db, Session session) {
+        TestContainerUtil.testCall(systemSession, APOC_DV_INSTALL_QUERY,
+                        getApocDvInstallParams(db),
+                        (row) -> assertCatalogContent(row, CSV_TEST_FILE));
+
+        session.executeRead(tx -> {
+                    final Result result = tx.run(APOC_DV_QUERY,
+                            Map.of(NAME_KEY, CSV_NAME_VALUE,
+                                    APOC_DV_QUERY_PARAMS_KEY, APOC_DV_QUERY_PARAMS,
+                                    CONFIG_KEY, CONFIG_VALUE)
+                    );
+
+                    Node node = result.single().get(NODE_KEY).asNode();
+                    assertEquals(NAME_VALUE, node.get(NAME_KEY).asString());
+                    assertEquals(AGE_VALUE, node.get(AGE_KEY).asString());
+                    assertEquals(List.of(LABELS_VALUE), node.labels());
+
+                    return result.consume();
+                }
+        );
+
+        String countDv = "CALL apoc.dv.catalog.show($db) YIELD name RETURN count(*) AS count";
+        long count = singleResultFirstColumn(systemSession, countDv, map("db", db));
+        assertEquals(1, count);
+
+        session.executeWrite(tx -> tx.run(CREATE_HOOK_QUERY, CREATE_HOOK_PARAMS).consume());
+
+        session.executeRead(tx -> {
+                    final Result result = tx.run(APOC_DV_QUERY_AND_LINK_QUERY,
+                            map(NAME_KEY, CSV_NAME_VALUE, APOC_DV_QUERY_PARAMS_KEY, APOC_DV_QUERY_PARAMS, RELTYPE_KEY, RELTYPE_VALUE, CONFIG_KEY, CONFIG_VALUE)
+                    );
+
+                    Path path = result.single().get("path").asPath();
+                    Node node = path.end();
+                    assertEquals(NAME_VALUE, node.get(NAME_KEY).asString());
+                    assertEquals(AGE_VALUE, node.get(AGE_KEY).asString());
+                    assertEquals(List.of(LABELS_VALUE), node.labels());
+
+                    Node hook = path.start();
+                    assertEquals(HOOK_NODE_NAME_VALUE, hook.get(NAME_KEY).asString());
+                    assertEquals(List.of("Hook"), hook.labels());
+
+                    Relationship relationship = path.relationships().iterator().next();
+                    assertEquals(hook.elementId(), relationship.startNodeElementId());
+                    assertEquals(node.elementId(), relationship.endNodeElementId());
+                    assertEquals(RELTYPE_VALUE, relationship.type());
+
+                    return result.consume();
+                }
+        );
+
+        testCallEmpty(systemSession, APOC_DV_DROP_QUERY, getApocDvDropParams(db));
+    }
+
+
+    private static void createDatabases() {
+        try(Session systemSession = driver.session(SessionConfig.forDatabase("system"))) {
+            systemSession.executeWrite(tx -> {
+                tx.run("CREATE DATABASE " + DB_FOO + " WAIT;").consume();
+                tx.run("CREATE ALIAS `test-alias` FOR DATABASE " + DB_FOO).consume();
+                return null;
+            });
+        }
+
+        try(Session systemSession = driver.session(SessionConfig.forDatabase("system"))) {
+            assertEventually(() -> {
+                final List<Record> list = systemSession.run("SHOW DATABASES YIELD name, currentStatus")
+                        .list();
+                return list.stream().allMatch(i -> i.get("currentStatus").asString().equals("online"))
+                        && list.stream().map(i -> i.get("name").asString()).toList().contains(DB_FOO);
+            }, value -> value, 30L, TimeUnit.SECONDS);
+        }
+    }
+
+    private static void createSessions() {
+        neo4jSession = neo4jContainer.getSession();
+        systemSession = driver.session(SessionConfig.forDatabase(SYSTEM_DATABASE_NAME));
+        fooSession = driver.session(SessionConfig.forDatabase(DB_FOO));
+    }
+}

--- a/extended-it/src/test/java/apoc/dv/DataVirtualizationCatalogTestUtil.java
+++ b/extended-it/src/test/java/apoc/dv/DataVirtualizationCatalogTestUtil.java
@@ -81,13 +81,15 @@ public class DataVirtualizationCatalogTestUtil {
     public static final String APOC_DV_LIST = "CALL apoc.dv.catalog.list()";
     public static final String APOC_DV_DROP_QUERY = "CALL apoc.dv.catalog.drop($name, $databaseName)";
     public static final String APOC_DV_INSTALL_QUERY = "CALL apoc.dv.catalog.install($name, $databaseName, $map)";
-    public static final Map<String, Object> APOC_DV_INSTALL_PARAMS = map(DATABASE_NAME, DEFAULT_DATABASE_NAME, 
-            NAME_KEY, CSV_NAME_VALUE, 
-            MAP_KEY, map("type", "CSV",
-                    "url", CSV_TEST_FILE, "query", QUERY_VALUE,
-                    "desc", DESC_VALUE,
-                    "labels", LABELS)
-    );
+    public static Map<String, Object> getApocDvInstallParams(String databaseName) {
+        return map(DATABASE_NAME, databaseName,
+                NAME_KEY, CSV_NAME_VALUE,
+                MAP_KEY, map("type", "CSV",
+                        "url", CSV_TEST_FILE, "query", QUERY_VALUE,
+                        "desc", DESC_VALUE,
+                        "labels", LABELS)
+        );
+    }
     public static final String APOC_DV_QUERY = "CALL apoc.dv.query($name, $queryParams, $config)";
     public static final String APOC_DV_QUERY_WITH_PARAM = "CALL apoc.dv.query($name, ['Italy'], $config)";
     public static final String APOC_DV_QUERY_AND_LINK_QUERY = "MATCH (hook:Hook) WITH hook " +
@@ -96,8 +98,10 @@ public class DataVirtualizationCatalogTestUtil {
     public static final String APOC_DV_SHOW_QUERY = "CALL apoc.dv.catalog.show()";
     
     public static Map<String, Object> APOC_DV_QUERY_PARAMS = map(NAME_KEY, NAME_VALUE, AGE_KEY, AGE_VALUE);
-    public static final Map<String, Object> APOC_DV_DROP_PARAMS = map(DATABASE_NAME, DEFAULT_DATABASE_NAME,
-            NAME_KEY, CSV_NAME_VALUE);
+    public static Map<String, Object> getApocDvDropParams(String databaseName) {
+        return map(DATABASE_NAME, databaseName,
+                NAME_KEY, CSV_NAME_VALUE);
+    }
 
     public static Map<String, Object> getJdbcCredentials(JdbcDatabaseContainer mysql) {
         return map("user", mysql.getUsername(), "password", mysql.getPassword());

--- a/extended-it/src/test/java/apoc/neo4j/docker/CustomNewProcedureMultiDbTest.java
+++ b/extended-it/src/test/java/apoc/neo4j/docker/CustomNewProcedureMultiDbTest.java
@@ -16,6 +16,7 @@ import java.util.Map;
 import java.util.concurrent.TimeUnit;
 
 import static apoc.util.ExtendedTestContainerUtil.singleResultFirstColumn;
+import static apoc.util.MapUtil.map;
 import static apoc.util.TestContainerUtil.*;
 import static java.util.concurrent.TimeUnit.SECONDS;
 import static org.junit.Assert.*;
@@ -93,6 +94,41 @@ public class CustomNewProcedureMultiDbTest {
         chackThatFunAndProcAreInstalledOnlyInTheSpecifiedDb(testSession, 
                 "CALL custom.testProc", "RETURN custom.testFun() AS answer",
                 fooSession, neo4jSession);
+    }
+
+    @Test
+    public void testProceduresFunctionsInDatabaseAlias() {
+        systemSession.executeWrite(tx -> tx.run("CREATE ALIAS `test-alias` FOR DATABASE dbfoo",
+                        Map.of("db", DB_TEST)).consume()
+        );
+
+        systemSession.executeWrite(tx ->
+                tx.run("CALL apoc.custom.installProcedure('testAliasProc() :: (answer::INT)','RETURN 42 as answer', 'test-alias')")
+                        .consume()
+        );
+        
+        systemSession.executeWrite(tx ->
+                tx.run("CALL apoc.custom.installFunction('testAliasFun() :: INT','RETURN 42 as answer', 'test-alias')")
+                        .consume()
+        );
+
+        chackThatFunAndProcAreInstalledOnlyInTheSpecifiedDb(fooSession,
+                "CALL custom.testAliasProc", 
+                "RETURN custom.testAliasFun() AS answer",
+                neo4jSession, testSession);
+
+        systemSession.executeWrite(tx -> {
+                tx.run("CALL apoc.custom.dropProcedure('testAliasProc', 'test-alias')")
+                        .consume();
+                tx.run("CALL apoc.custom.dropFunction('testAliasFun', 'test-alias')")
+                        .consume();
+                return null;
+        });
+
+        String countCustom = "CALL apoc.custom.show('test-alias') YIELD name RETURN count(*) AS count";
+        long count = singleResultFirstColumn(systemSession, countCustom);
+        assertEquals(0, count);
+        
     }
 
     private static void installTestProcAndFun() {

--- a/extended-it/src/test/java/apoc/neo4j/docker/UUIDMultiDbTest.java
+++ b/extended-it/src/test/java/apoc/neo4j/docker/UUIDMultiDbTest.java
@@ -21,6 +21,8 @@ import java.util.stream.Stream;
 
 import static apoc.ExtendedApocConfig.APOC_UUID_ENABLED;
 import static apoc.ExtendedApocConfig.APOC_UUID_ENABLED_DB;
+import static apoc.util.ExtendedTestContainerUtil.singleResultFirstColumn;
+import static apoc.util.MapUtil.map;
 import static apoc.util.TestContainerUtil.createEnterpriseDB;
 import static apoc.uuid.UUIDTestUtils.assertIsUUID;
 import static apoc.uuid.UuidHandler.APOC_UUID_REFRESH;
@@ -182,6 +184,44 @@ public class UUIDMultiDbTest {
                     .get("uuid");
             assertTrue(uuid.isNull());
         }
+    }
+
+    @Test
+    public void setupUUIDUsingAliasInNotDefaultDbWithUUIDEnabled() throws InterruptedException {
+        driver.session(SYS_CONF).executeWrite(tx -> tx.run("CREATE ALIAS `test-alias` FOR DATABASE " + DB_ENABLED)
+                .consume()
+        );
+
+        driver.session(SYS_CONF)
+                .executeWrite(tx -> tx.run("CALL apoc.uuid.setup('Alias', $db, {addToExistingNodes: false })",
+                        Map.of("db", "test-alias")).consume()
+                );
+
+        try(Session session = driver.session(SessionConfig.forDatabase(DB_ENABLED))) {
+            // check uuid set
+            awaitUuidSet(session, "Alias");
+
+            session.executeWrite(tx -> tx.run("CREATE (:Alias)").consume());
+
+            Result res = session.run("MATCH (n:Alias) RETURN n.uuid AS uuid");
+            String uuid = res.single().get("uuid").asString();
+            assertIsUUID(uuid);
+        }
+
+        try(Session session = driver.session()) {
+            session.run("CREATE (:Alias)");
+            // we cannot use assertEventually because the before and after conditions should be the same
+            Thread.sleep(2000);
+
+            Value uuid = session.run("MATCH (n:Alias) RETURN n.uuid AS uuid")
+                    .single()
+                    .get("uuid");
+            assertTrue(uuid.isNull());
+        }
+
+        String countUUid = "CALL apoc.uuid.show($db) YIELD label RETURN count(*) AS count";
+        long count = singleResultFirstColumn(driver.session(SYS_CONF), countUUid, map("db", "test-alias"));
+        assertEquals(0, count);
     }
 
     private static void awaitUuidSet(Session session, String expected) {

--- a/extended/src/main/java/apoc/custom/CypherNewProcedures.java
+++ b/extended/src/main/java/apoc/custom/CypherNewProcedures.java
@@ -15,6 +15,7 @@ import java.util.stream.Stream;
 import static apoc.custom.CypherProceduresHandler.PREFIX;
 import static apoc.custom.CypherProceduresUtil.checkEnabled;
 import static apoc.util.SystemDbUtil.checkInSystemDb;
+import static apoc.util.SystemDbUtil.getDbFromDbNameOrAliasForReadProcedures;
 
 
 @Extended
@@ -29,12 +30,12 @@ public class CypherNewProcedures {
     @Context
     public Transaction tx;
 
-    private void checkIsValidDatabase(String databaseName) {
+    private String checkIsValidDatabase(String databaseName) {
         SystemDbUtil.checkInSystemLeader(db);
 
-        SystemDbUtil.checkTargetDatabase(tx, databaseName, "Custom procedures/functions");
-
         checkEnabled();
+
+        return SystemDbUtil.checkTargetDatabase(tx, databaseName, "Custom procedures/functions");
     }
 
     // TODO - change with @SystemOnlyProcedure
@@ -47,12 +48,12 @@ public class CypherNewProcedures {
                                  @Name(value = "databaseName", defaultValue = "neo4j") String databaseName,
                                  @Name(value = "mode", defaultValue = "read") String mode,
                                  @Name(value = "description", defaultValue = "") String description) {
-        checkIsValidDatabase(databaseName);
+        String dbNameOrAlias = checkIsValidDatabase(databaseName);
 
         Mode modeProcedure = CypherProceduresUtil.mode(mode);
         ProcedureSignature procedureSignature = new Signatures(PREFIX).asProcedureSignature(signature, description, modeProcedure);
 
-        CypherHandlerNewProcedure.installProcedure(databaseName, procedureSignature, statement);
+        CypherHandlerNewProcedure.installProcedure(dbNameOrAlias, procedureSignature, statement);
     }
 
     // TODO - change with @SystemOnlyProcedure
@@ -64,10 +65,10 @@ public class CypherNewProcedures {
                                 @Name(value = "databaseName", defaultValue = "neo4j") String databaseName,
                                 @Name(value = "forceSingle", defaultValue = "false") boolean forceSingle,
                                 @Name(value = "description", defaultValue = "") String description) {
-        checkIsValidDatabase(databaseName);
+        String dbNameOrAlias = checkIsValidDatabase(databaseName);
 
         UserFunctionSignature userFunctionSignature = new Signatures(PREFIX).asFunctionSignature(signature, description);
-        CypherHandlerNewProcedure.installFunction(databaseName, userFunctionSignature, statement, forceSingle);
+        CypherHandlerNewProcedure.installFunction(dbNameOrAlias, userFunctionSignature, statement, forceSingle);
     }
 
     // TODO - change with @SystemOnlyProcedure
@@ -76,9 +77,9 @@ public class CypherNewProcedures {
     @Procedure(value = "apoc.custom.dropProcedure", mode = Mode.WRITE)
     @Description("Eventually drops the targeted custom procedure")
     public void dropProcedure(@Name("name") String name, @Name(value = "databaseName", defaultValue = "neo4j") String databaseName) {
-        checkIsValidDatabase(databaseName);
+        String dbNameOrAlias = checkIsValidDatabase(databaseName);
 
-        CypherHandlerNewProcedure.dropProcedure(databaseName, name);
+        CypherHandlerNewProcedure.dropProcedure(dbNameOrAlias, name);
     }
 
     // TODO - change with @SystemOnlyProcedure
@@ -87,9 +88,9 @@ public class CypherNewProcedures {
     @Procedure(value = "apoc.custom.dropFunction", mode = Mode.WRITE)
     @Description("Eventually drops the targeted custom function")
     public void dropFunction(@Name("name") String name, @Name(value = "databaseName", defaultValue = "neo4j") String databaseName) {
-        checkIsValidDatabase(databaseName);
+        String dbNameOrAlias = checkIsValidDatabase(databaseName);
 
-        CypherHandlerNewProcedure.dropFunction(databaseName, name);
+        CypherHandlerNewProcedure.dropFunction(dbNameOrAlias, name);
     }
 
     // TODO - change with @SystemOnlyProcedure
@@ -98,9 +99,9 @@ public class CypherNewProcedures {
     @Procedure(value = "apoc.custom.dropAll", mode = Mode.WRITE)
     @Description("Eventually drops all previously added custom procedures/functions and returns info")
     public Stream<CustomProcedureInfo> dropAll(@Name(value = "databaseName", defaultValue = "neo4j") String databaseName) {
-        checkIsValidDatabase(databaseName);
+        String dbNameOrAlias = checkIsValidDatabase(databaseName);
 
-        return CypherHandlerNewProcedure.dropAll(databaseName)
+        return CypherHandlerNewProcedure.dropAll(dbNameOrAlias)
                 .stream();
     }
 
@@ -112,6 +113,7 @@ public class CypherNewProcedures {
     @Description("Provides a list of custom procedures/function registered")
     public Stream<CustomProcedureInfo> show(@Name(value = "databaseName", defaultValue = "neo4j") String databaseName) {
         checkInSystemDb(db);
+        String dbNameOrAlias = getDbFromDbNameOrAliasForReadProcedures(tx, databaseName, db);
         return CypherHandlerNewProcedure.show(databaseName, tx);
     }
 }

--- a/extended/src/main/java/apoc/dv/DataVirtualizationCatalogNewProcedures.java
+++ b/extended/src/main/java/apoc/dv/DataVirtualizationCatalogNewProcedures.java
@@ -25,9 +25,9 @@ public class DataVirtualizationCatalogNewProcedures {
     @Context
     public GraphDatabaseAPI db;
 
-    private void checkIsValidDatabase(String databaseName) {
+    private String checkIsValidDatabase(String databaseName) {
         SystemDbUtil.checkInSystemLeader(db);
-        SystemDbUtil.checkTargetDatabase(tx, databaseName, "Data virtualization catalog");
+        return SystemDbUtil.checkTargetDatabase(tx, databaseName, "Data virtualization catalog");
     }
 
     @SystemProcedure
@@ -38,10 +38,10 @@ public class DataVirtualizationCatalogNewProcedures {
             @Name("name") String name,
             @Name(value = "databaseName", defaultValue = "neo4j") String databaseName,
             @Name(value = "config", defaultValue = "{}") Map<String,Object> config) {
-        checkIsValidDatabase(databaseName);
+        String dbNameOrAlias = checkIsValidDatabase(databaseName);
 
         return Stream.of(new DataVirtualizationCatalogHandlerNewProcedures()
-                        .install(databaseName, VirtualizedResource.from(name, config)))
+                        .install(dbNameOrAlias, VirtualizedResource.from(name, config)))
                 .map(VirtualizedResource::toDTO);
     }
 
@@ -53,9 +53,9 @@ public class DataVirtualizationCatalogNewProcedures {
             @Name("name") String name,
             @Name(value = "databaseName", defaultValue = "neo4j") String databaseName
     ) {
-        checkIsValidDatabase(databaseName);
+        String dbNameOrAlias = checkIsValidDatabase(databaseName);
         return new DataVirtualizationCatalogHandlerNewProcedures()
-                .drop(databaseName, name)
+                .drop(dbNameOrAlias, name)
                 .map(VirtualizedResource::toDTO);
     }
 
@@ -65,9 +65,9 @@ public class DataVirtualizationCatalogNewProcedures {
     public Stream<VirtualizedResource.VirtualizedResourceDTO> show(
             @Name(value = "databaseName", defaultValue = "neo4j") String databaseName
     ) {
-        checkIsValidDatabase(databaseName);
+        String dbNameOrAlias = checkIsValidDatabase(databaseName);
         return new DataVirtualizationCatalogHandlerNewProcedures()
-                .show(databaseName)
+                .show(dbNameOrAlias)
                 .map(VirtualizedResource::toDTO);
     }
 

--- a/extended/src/main/java/apoc/util/SystemDbUtil.java
+++ b/extended/src/main/java/apoc/util/SystemDbUtil.java
@@ -8,13 +8,9 @@ import org.neo4j.graphdb.ResourceIterator;
 import org.neo4j.graphdb.Transaction;
 import org.neo4j.kernel.internal.GraphDatabaseAPI;
 
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 import java.util.function.Consumer;
 import java.util.function.Function;
-import java.util.stream.Collectors;
 
 import static apoc.ApocConfig.apocConfig;
 import static apoc.SystemPropertyKeys.database;
@@ -66,23 +62,42 @@ public class SystemDbUtil {
     }
 
     /**
-     * Check that the database exists and is not equal to "system"
+     * Check that the database name or alias exists and is not equal to "system"
      *
      * @param databaseName
      * @param type: the procedure type
+     * @return databaseName
      */
-    public static void checkTargetDatabase(Transaction tx, String databaseName, String type) {
-        final Set<String> databases = tx.execute("SHOW DATABASES", Collections.emptyMap())
-                .<String>columnAs("name")
-                .stream()
-                .collect(Collectors.toSet());
-        if (!databases.contains(databaseName)) {
-            throw new RuntimeException( String.format(DB_NOT_FOUND_ERROR, databaseName) );
-        }
-
+    public static String checkTargetDatabase(Transaction tx, String databaseName, String type) {
         if (databaseName.equals(SYSTEM_DATABASE_NAME)) {
             throw new RuntimeException(type + BAD_TARGET_ERROR);
         }
+
+        return getDbFromDbNameOrAlias(tx, databaseName);
+    }
+
+    public static String getDbFromDbNameOrAliasForReadProcedures(Transaction tx, String databaseName, GraphDatabaseService db) {
+        if (!SYSTEM_DATABASE_NAME.equals(db.databaseName())) {
+            // If we are not executing a procedure against the systemdb we cannot retrieve the aliases
+            // since `SHOW DATABASES` is a system command, so we just get the `databaseName` 
+            return databaseName;
+        }
+        
+        return getDbFromDbNameOrAlias(tx, databaseName);
+    }
+
+    public static String getDbFromDbNameOrAlias(Transaction tx, String databaseName) {
+
+        
+        var databasesAndAliases = tx.execute("SHOW DATABASES", Collections.emptyMap()).stream()
+                .map(i -> new AbstractMap.SimpleEntry<>((String) i.get("name"), (List<String>) i.get("aliases")))
+                .toList();
+
+        String database = databasesAndAliases.stream().filter(i -> i.getKey().equals(databaseName) || i.getValue().contains(databaseName))
+                .findFirst()
+                .map(AbstractMap.SimpleEntry::getKey)
+                .orElseThrow(() -> new RuntimeException(String.format(DB_NOT_FOUND_ERROR, databaseName)));
+        return database;
     }
 
     /**

--- a/extended/src/main/java/apoc/uuid/UUIDNewProcedures.java
+++ b/extended/src/main/java/apoc/uuid/UUIDNewProcedures.java
@@ -13,6 +13,7 @@ import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import static apoc.ApocConfig.apocConfig;
+import static apoc.util.SystemDbUtil.getDbFromDbNameOrAliasForReadProcedures;
 import static apoc.util.SystemDbUtil.withSystemDb;
 import static apoc.uuid.UUIDHandlerNewProcedures.checkEnabled;
 import static apoc.uuid.UuidHandler.APOC_UUID_REFRESH;
@@ -27,15 +28,13 @@ public class UUIDNewProcedures {
     @Context
     public Transaction tx;
 
-    private void checkInSystemLeader(String databaseName) {
+    private String checkInSystemLeader(String databaseName) {
         checkEnabled(databaseName);
         checkRefreshConfigSet();
 
         SystemDbUtil.checkInSystemLeader(db);
-    }
-
-    private void checkTargetDatabase(String databaseName) {
-        SystemDbUtil.checkTargetDatabase(tx, databaseName, "Automatic UUIDs");
+        
+        return SystemDbUtil.checkTargetDatabase(tx, databaseName, "Automatic UUIDs");
     }
 
     private void checkRefreshConfigSet() {
@@ -52,13 +51,12 @@ public class UUIDNewProcedures {
     public Stream<UuidInfo> setup(@Name("label") String label,
                                   @Name(value = "databaseName", defaultValue = "neo4j") String databaseName,
                                   @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
-        checkInSystemLeader(databaseName);
-        checkTargetDatabase(databaseName);
+        String dbNameOrAlias = checkInSystemLeader(databaseName);
 
         UuidConfig uuidConfig = new UuidConfig(config);
 
         // unlike the apoc.uuid.install we don't return the UuidInstallInfo because we don't retrieve the `batchComputationResult` field
-        UuidInfo uuidInfo = UUIDHandlerNewProcedures.create(databaseName, label, uuidConfig);
+        UuidInfo uuidInfo = UUIDHandlerNewProcedures.create(dbNameOrAlias, label, uuidConfig);
         return Stream.of(uuidInfo);
     }
 
@@ -68,9 +66,9 @@ public class UUIDNewProcedures {
     @Procedure(mode = Mode.WRITE)
     @Description("CALL apoc.uuid.drop(label, databaseName) yield label, installed, properties | eventually removes previously added UUID handler and returns uuid information")
     public Stream<UuidInfo> drop(@Name("label") String label, @Name(value = "databaseName", defaultValue = "neo4j") String databaseName) {
-        checkInSystemLeader(databaseName);
+        String dbNameOrAlias = checkInSystemLeader(databaseName);
 
-        final UuidInfo uuidInfo = UUIDHandlerNewProcedures.drop(databaseName, label);
+        final UuidInfo uuidInfo = UUIDHandlerNewProcedures.drop(dbNameOrAlias, label);
         return Stream.ofNullable(uuidInfo);
     }
 
@@ -80,9 +78,9 @@ public class UUIDNewProcedures {
     @Procedure(mode = Mode.WRITE)
     @Description("CALL apoc.uuid.dropAll(databaseName) yield label, installed, properties | eventually removes all previously added UUID handlers and returns uuids' information")
     public Stream<UuidInfo> dropAll(@Name(value = "databaseName", defaultValue = "neo4j") String databaseName) {
-        checkInSystemLeader(databaseName);
+        String dbNameOrAlias = checkInSystemLeader(databaseName);
 
-        return UUIDHandlerNewProcedures.dropAll(databaseName)
+        return UUIDHandlerNewProcedures.dropAll(dbNameOrAlias)
                 .stream()
                 .sorted(Comparator.comparing(i -> i.label));
     }
@@ -95,6 +93,7 @@ public class UUIDNewProcedures {
     @Description("CALL apoc.uuid.show(databaseName) | it lists all eventually installed UUID handler for a database")
     public Stream<UuidInfo> show(@Name(value = "databaseName", defaultValue = "neo4j") String databaseName) {
         checkEnabled(databaseName);
+        String dbNameOrAlias = getDbFromDbNameOrAliasForReadProcedures(tx, databaseName, db);
 
         return withSystemDb(sysTx -> {
             return UUIDHandlerNewProcedures.getUuidNodes(sysTx, databaseName)

--- a/extended/src/test/java/apoc/util/ExtendedTestContainerUtil.java
+++ b/extended/src/test/java/apoc/util/ExtendedTestContainerUtil.java
@@ -32,7 +32,11 @@ public class ExtendedTestContainerUtil
     }
 
     public static <T> T singleResultFirstColumn(Session session, String cypher) {
-        return (T) session.executeWrite(tx -> tx.run(cypher).single().fields().get(0).value().asObject());
+        return singleResultFirstColumn(session, cypher, Map.of());
+    }
+    
+    public static <T> T singleResultFirstColumn(Session session, String cypher, Map<String, Object> params) {
+        return (T) session.executeWrite(tx -> tx.run(cypher, params).single().fields().get(0).value().asObject());
     }
 
     public static void testCallInReadTransaction(Session session, String call, Consumer<Map<String, Object>> consumer) {

